File: trunk/trafficlight.cpp
#include "trafficlight.h"

TrafficLight::TrafficLight()
{
}

void TrafficLight::setPoint(QPoint centro)
{
    this->centro = centro;
}

void TrafficLight::setPoint(int x, int y)
{
    this->centro.setX(x);
    this->centro.setY(y);
}

void TrafficLight::setX(int x)
{
    this->centro.setX(x);
}

void TrafficLight::setY(int y)
{
    this->centro.setY(y);
}

void TrafficLight::setColor(QColor color)
{
    this->color = color;
}

void TrafficLight::setCircle(QPoint tamano)
{
    this->tamano = tamano;
}

void TrafficLight::setCircle(int ancho, int alto)
{
    this->tamano.setX(ancho);
    this->tamano.setY(alto);
}

void TrafficLight::setTiempo(int tiempo)
{
    this->tiempo = tiempo;
}

void TrafficLight::setOn(bool on)
{
    this->on = on;
}

QPoint TrafficLight::getPoint()
{
    return centro;
}

int TrafficLight::getX()
{
    return centro.x();
}

int TrafficLight::getY()
{
    return centro.y();
}

QColor TrafficLight::getColor()
{
    return color;
}

QPoint TrafficLight::getCircle()
{
    return tamano;
}

int TrafficLight::getTiempo()
{
    return tiempo;
}

bool TrafficLight::getIson()
{
    return on;
}

File: SemaforoInteligente/src/com/ssi/object/TrafficLight.java
package com.ssi.object;

import java.awt.Color;

public class TrafficLight {

	private int x;
	private int y;
	private int rX;
	private int rY;
	private Color color;
	private int tiempo; /*seconds*/
	private boolean on;	
	
	public TrafficLight()
	{
		
	}

	public TrafficLight(int x, int y, int rX, int rY, Color color, int tiempo, boolean on)
	{
		this.x = x;
		this.y = y;
		this.rX = rX;
		this.rY = rY;
		this.color = color;
		this.tiempo = tiempo;
		this.on = on;
	}
	
	public int getX() {
		return x;
	}
	
	public void setX(int x) {
		this.x = x;
	}
	
	public int getY() {
		return y;
	}
	
	public void setY(int y) {
		this.y = y;
	}
	
	public int getrX() {
		return rX;
	}

	public void setrX(int rX) {
		this.rX = rX;
	}

	public int getrY() {
		return rY;
	}

	public void setrY(int rY) {
		this.rY = rY;
	}
	
	public Color getColor() {
		return color;
	}
	
	public void setColor(Color color) {
		this.color = color;
	}
	
	public int getTiempo() {
		return tiempo;
	}
	
	public void setTiempo(int tiempo) {
		this.tiempo = tiempo;
	}
	
	public boolean isOn() {
		return on;
	}
	
	public void setOn(boolean on) {
		this.on = on;
	}

}
